aiida/orm/nodes/node.py

```diff
@@ -328,67 +368,187 @@ def mtime(self):
+            warnings.warn(
+                'keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.',
+                AiidaDeprecationWarning
+            )  # pylint: disable=no-member
+            path = key
+
+        return self._repository.list_objects(path)
+
+    def list_object_names(self, path=None, key=None):
         """Return a list of the object names contained in this repository, optionally in the given sub directory.
 
+        .. deprecated:: 1.4.0
+            Keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.
+
+        :param path: the relative path of the object within the repository.
         :param key: fully qualified identifier for the object within the repository
-        :return: a list of `File` named tuples representing the objects present in directory with the given key
+        :return: a list of `File` named tuples representing the objects present in directory with the given path
         """
-        return self._repository.list_object_names(key)
+        if key is not None:
+            if path is not None:
+                raise ValueError('cannot specify both `path` and `key`.')
+            warnings.warn(
+                'keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.',
+                AiidaDeprecationWarning
+            )  # pylint: disable=no-member
+            path = key
+
+        return self._repository.list_object_names(path)
 
-    def open(self, key, mode='r'):
-        """Open a file handle to an object stored under the given key.
+    def open(self, path=None, mode='r', key=None):
+        """Open a file handle to the object with the given path.
 
+        .. deprecated:: 1.4.0
+            Keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.
+
+        :param path: the relative path of the object within the repository.
         :param key: fully qualified identifier for the object within the repository
         :param mode: the mode under which to open the handle
         """
-        return self._repository.open(key, mode)
+        if key is not None:
+            if path is not None:
+                raise ValueError('cannot specify both `path` and `key`.')
+            warnings.warn(
+                'keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.',
+                AiidaDeprecationWarning
+            )  # pylint: disable=no-member
+            path = key
+
+        if path is None:
+            raise TypeError("open() missing 1 required positional argument: 'path'")
+
+        if mode not in ['r', 'rb']:
+            warnings.warn("from v2.0 only the modes 'r' and 'rb' will be accepted", AiidaDeprecationWarning)  # pylint: disable=no-member
 
-    def get_object(self, key):
-        """Return the object identified by key.
+        return WarnWhenNotEntered(self._repository.open(path, mode), repr(self))
```

Comment:
Naively I'd have put the `WarnWhenNotEntered` mechanism at the level of `Repository.open` - but if only `Node.open` is a public interface, maybe this is fine also.
I'll leave that up to you to decide @sphuber.

Reply:
The `Repository` is not public interface. If users do use that, they will get another deprecation warning first, so they will eventually also see the `open` warning